File: Anagrams/Anagrams.cpp
class Solution {
public:
    vector<string> anagrams(vector<string> &strs) {
        unordered_map<string, vector<string>> map;
        for (int i = 0; i < strs.size(); ++i) {
            string s = string(strs[i]);
            sort(s.begin(), s.end());
            auto it = map.find(s);
            if (it == map.end()) {
                vector<string> list;
                list.push_back(strs[i]);
                map[s] = list;
            } else {
                it->second.push_back(strs[i]);
            }
        }
        vector<string> res;
        for (auto it = map.begin(); it != map.end(); ++it) {
            if (it->second.size() > 1) {
                for (string s : it->second) {
                    res.push_back(s);
                }
            }
        }
        return res;
    }
};

File: Anagrams/Anagrams.java
public class Solution {
    public List<String> anagrams(String[] strs) {
        HashMap<String, List<String>> map = new HashMap<>();
        for (int i = 0; i < strs.length; ++i) {
            char [] c = strs[i].toCharArray();
            Arrays.sort(c);
            String s = new String(c);
            List<String> list = map.get(s);
            if (list == null) {
                list = new ArrayList<String>();
                map.put(s, list);
            }
            list.add(strs[i]);
        }
        List<String> res = new ArrayList<String>();
        for (String key : map.keySet()) {
            List<String> l = map.get(key);
            if (l.size() > 1) {
                for (String s : l) {
                    res.add(s);
                }
            }
        }
        return res;
    }
}